tensorflow/compiler/xla/literal.cc

```diff
@@ -1891,13 +1894,13 @@ bool LiteralBase::IsR1Iota() const {
   auto is_iota_at_idx = [&](const int64 idx) {
     switch (shape().element_type()) {
       case U8:
-        return Get<uint8>({idx}) == idx;
+        return Get<uint8>({idx}) == static_cast<uint8>(idx);
       case U16:
-        return Get<uint16>({idx}) == idx;
+        return Get<uint16>({idx}) == static_cast<uint16>(idx);
       case U32:
-        return Get<uint32>({idx}) == idx;
+        return Get<uint32>({idx}) == static_cast<uint32>(idx);
       case U64:
-        return Get<uint64>({idx}) == idx;
+        return Get<uint64>({idx}) == static_cast<uint64>(idx);
```

Comment:
https://godbolt.org/z/sxThad says that we should instead `static_cast<int64>(Get<...>)` instead of the RHS.

Didn't notice the bug until now, this might be why builds fail